Tamarin Prover 1.2.1 (new formula)

mpdehnel opened the pull request:
Adding the Tamarin Prover (new formula, 1.2.1) to homebrew/science.

This was previously in Homebrew Cask as a binary (in versions 1.0.0 and 1.2.0), but now that various compilation and stable release cycle issues have been sorted out, it's ready and suitable to go
into the main tap of Homebrew Science.
I will ask Cask to remove it as soon as it is accepted into Homebrew Science.
Cask would prefer it to be in non-Cask Homebrew, as it is an open-source project.
Please see [the original pull request](https://github.com/caskroom/homebrew-cask/pull/24273) adding it to Cask for details!

### Have you:

- [x] Followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew-science/blob/master/.github/CONTRIBUTING.md) document?
- [x] Checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew-science/pulls) for the same update/change?
- [x] Run `brew audit --strict --online <formula>` (where `<formula>` is the name of the formula you're submitting) and corrected all errors?
- [x] Built your formula locally prior to submission with `brew install <formula>`?

### New formula: have you

- [x] Written a sensible test? (The best test is to compile and run a sample program.)
Tamarin Prover has a `tamarin-prover test` command: returning 0 on it means it passed all the self-tests. This is what the brew file runs and checks. This passes on my local install.

### Updates to existing formula: have you

- [ ] Removed the `revision` line, if any, when bumping the version number?
- [ ] Added/bumped the `revision` number if the changes affect the precompiled bottles?
- [ ] Not altered the `bottle` section?
- [ ] Checked that the tests still pass?

lordqwerty commented:
I would like to propose an exception for the suggested formula is made. This is a great piece of software that would really help the developers to promote their software, which will be enabled through the ease of running `brew install`. 

In my personal opinion it will only be a matter of time until the Tamarin Prover will meet the criteria, as within computer science academia the tool is attracting a lot of interest. Homebrew Science is the perfect place for the tool and can help in so many ways by accepting this PR.

As a user of the tool I can personally say from experience that when a bug or issue has been spotted with Tamarin they are extremely reactive and have a fix ready / respond to your issues. 

Many thanks for taking the time to read this.
I look forward to hearing your decision. 
All the best,

Jorden

mpdehnel commented:
I am slightly confused why this is failing the build: if the only issue is notability, then I refer you to Jorden's comments above.

If not, I would appreciate advice: all build tests and commands build successfully on Sierra for both myself and @katrielalex. If there are specific issues on El Capitan, we would obviously like to address those, or at worst, require Sierra.

jonchang commented on tamarin-prover.rb:
```diff
@@ -0,0 +1,25 @@
+      :revision => "596eba95cd00e1daa7db7d054d42d6790f96fcda"
+  sha256 "9ff2dfa18034427fa9047b9bad43946c2d9212031da7535434d6ac8cc16e69f0"
+  head "https://github.com/tamarin-prover/tamarin-prover.git", :branch => "develop"
+
```
Please add a `# doi` and `# tag` after `head`. If you need some examples try `cd $(brew --repo homebrew/science) && grep "# doi" *`

mpdehnel replied:
Added.

mpdehnel commented:
Hi Jon,

The issue as far as I can see is that Stack is set up to only build the bottle on El Capitan and earlier to get around a GHC bug before version 8.0.2. So two possible solutions: set Tamarin up the same way, and use the El Capitan binary in Sierra (which will work fine), or fix the Stack formula which is the proper solution. Except Stack by default builds with an old GHC, they have experimental support for 8.0.2 bootstrapping but it's not the default.

Having discussed it with other people working on the active development of Tamarin, I think it would be best to leave it as a cask until Stack bootstraps with GHC 8.0.2 (that way we can do it right first time, and it's not urgent to get it into homebrew/science).

We'll be back once this issue is resolved, and do it right :-)

Thanks anyway!

Martin

mpdehnel commented on tamarin-prover.rb:
```diff
@@ -0,0 +1,25 @@
+  url "https://github.com/tamarin-prover/tamarin-prover",
+      :using => :git,
+      :tag => "1.2.1",
+      :revision => "596eba95cd00e1daa7db7d054d42d6790f96fcda"
```
Changed.

mpdehnel commented on tamarin-prover.rb:
```diff
@@ -0,0 +1,25 @@
+
+  depends_on "haskell-stack" => :build
+  depends_on "homebrew/science/maude" => :run
+  depends_on "graphviz" => :run
```
Sorted.

mpdehnel commented on tamarin-prover.rb:
```diff
@@ -0,0 +1,25 @@
+  def install
+    system "make", "tamarin"
+
+    bin.install ".brew_home/.local/bin/tamarin-prover"
```
We tried it with `bin.install "tamarin-prover"` but I don't think it worked.

katrielalex replied:
Stack installs to `~/.local/bin`, which isn't on the PATH in the homebrew shell. There might be an incantation to put it on the PATH, if you prefer that to specifying it here.

mpdehnel commented:
Re-opened, because with the updates to Stack, this should (hopefully) now build. Have also addressed your comments above -- please let me know if there are any other changes needed!

mpdehnel commented:
Glad to see it now builds correctly on El Capitan, Sierra, and Yosemite!

jonchang commented:
Looks like Linux bottles are timing out due to long build times.

jonchang commented:
Thanks for your hard work @mpdehnel! This has been merged.
